src/CoreWCF.Http/tests/Authorization/AuthorizationPoliciesTests.cs

```diff
@@ -0,0 +1,258 @@
+                                    $"<{operationContractName}Result>A</{operationContractName}Result>" +
+                                    $"</{operationContractName}Response>" +
+                                    "</s:Body>" +
+                                    "</s:Envelope>";
+
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.Equal(expected, responseBody);
+        }
+
+        public class Startup
+        {
+            public void ConfigureServices(IServiceCollection services)
+            {
+                services.AddAuthorization(options =>
+                {
+                    options.AddPolicy(Policies.Write,
+                        policy => policy.RequireClaim("scope", new[] { DefinedScopes.Write, DefinedScopes.Admin }));
+                    options.AddPolicy(Policies.AdminOnly,
+                        policy => policy.RequireClaim("scope", new[] { DefinedScopes.Admin }));
+                });
+                services.AddServiceModelServices();
+                services.AddSingleton<IServiceBehavior, AuthorizationServiceBehavior>();
+            }
+
+            public void Configure(IApplicationBuilder app)
+            {
+                app.UseServiceModel(builder =>
+                {
+                    builder.AddService<EchoService>();
+                    builder.AddServiceEndpoint<EchoService, IEchoService>(
+                        new BasicHttpBinding
+                        {
+                            Security = new BasicHttpSecurity
+                            {
+                                Transport = new HttpTransportSecurity
+                                {
+                                    ClientCredentialType = HttpClientCredentialType.Custom,
+                                    CustomAuthenticationScheme = "Bearer"
+                                }
+                            }
+                        }, "/BasicWcfService/basichttp.svc");
+                });
+            }
+        }
+
+        [ServiceContract]
+        public interface IEchoService
+        {
+            [OperationContract]
+            string Default(string text);
+            [OperationContract]
+            string AdminOnly(string text);
+            [OperationContract]
+            string Anonymous(string text);
+            [OperationContract]
+            Task<string> Write(string text);
+        }
+
+        [ServiceBehavior(IncludeExceptionDetailInFaults = true)]
+        public class EchoService : IEchoService
+        {
+            // No attribute => defaults to the builtin default policy which is RequireAuthenticatedUser
+            public string Default(string text) => text;
+            [Authorize(Policy = Policies.AdminOnly)]
+            public string AdminOnly(string text) => text;
+            [AllowAnonymous]
+            public string Anonymous(string text) => text;
```

Comment:
I don't think we can support the AllowAnonymous attribute. The granularity of specifying authentication type is at the service endpoint. If you have authentication required on the binding of an endpoint, it will require all requests for that endpoint to be authenticated. The way you allow some methods to be allowed to be called anonymously is you define those methods on a separate contract interface which only contains the anonymous methods and add another endpoint with the binding configured for anonymous authentication.  
We can't know which operation is being called at the transport level because the operation is selected using an IOperationSelector which is the second from last extensibility point used in the ImmutableDispatchRuntime (with IOperationInvoker being the last). There's just no way to know which operation is being called at the transport layer where authentication is being requested.  
I'll explain in a separate comment on the code where you are checking the properties on the returned message about the issues with that approach.

Reply:
I dropped Anonymous support